torch/csrc/jit/codegen/cuda/parser.cpp

```diff
@@ -694,6 +692,191 @@ class IrParser {
+            if (!node->input(5)->type()->isSubtypeOf(
+                    static_cast<c10::TypePtr>(NoneType::get()))) {
+              // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)
+              weight = value_map[node->input(5)->unique()]->as<TensorView>();
+            }
+
+            TensorView* bias = nullptr;
+            // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)
+            if (!node->input(6)->type()->isSubtypeOf(
+                    static_cast<c10::TypePtr>(NoneType::get()))) {
+              // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)
+              bias = value_map[node->input(6)->unique()]->as<TensorView>();
+            }
+
+            const int kNormShapeNumDims = norm_shape->vec().size();
+            const size_t kOuterNumDims = input->nDims() - kNormShapeNumDims;
+
+            std::vector<int> outer_reduction_axes(kOuterNumDims);
+            std::vector<bool> outer_broadcast_mask(input->nDims(), false);
+            for (size_t idx = 0; idx < kOuterNumDims; ++idx) {
+              outer_reduction_axes[idx] = idx;
+              outer_broadcast_mask[idx] = true;
+            }
+
+            std::vector<int> inner_reduction_axes(kNormShapeNumDims);
+            std::vector<bool> inner_broadcast_mask(input->nDims(), false);
+            Val* num_features = new Double(1);
+            for (int idx = 0; idx < kNormShapeNumDims; ++idx) {
+              const size_t axis = input->nDims() - 1 - idx;
+              inner_reduction_axes[idx] = axis;
+              inner_broadcast_mask[axis] = true;
+              num_features =
+                  mul(num_features, input->domain()->domain()[axis]->extent());
+            }
+
+            // TODO: grad_bias and grad_weight are disabled because
+            // they are incompabilble with grad_in fusion
+            // Requires seperate kernels
+
+            auto x_hat = mul(sub(input, mean), rstd);
+
+            TensorView* grad_x_hat = nullptr;
+            if (weight != nullptr) {
+              auto* bcast_weight = broadcast(weight, outer_broadcast_mask);
+              grad_x_hat = mul(grad_out, bcast_weight);
+            } else {
+              grad_x_hat = grad_out;
+            }
+
+            auto* a = mul(num_features, grad_x_hat);
+
+            auto* b = sum(grad_x_hat, inner_reduction_axes);
+            auto* bcast_b = broadcast(b, inner_broadcast_mask);
+
+            auto* c1 = mul(grad_x_hat, x_hat);
+            auto* c2 = sum(c1, inner_reduction_axes);
+            auto* bcast_c2 = broadcast(c2, inner_broadcast_mask);
+            auto* c3 = mul(x_hat, bcast_c2);
+
+            auto* inner = sub(sub(a, bcast_b), c3);
+
+            auto reciprocal_size =
+                unaryOp(UnaryOpType::Reciprocal, num_features);
+            auto* grad_in = mul(mul(reciprocal_size, rstd), inner);
+
+            value_map.emplace(node->output(0)->unique(), grad_in);
+
+            // if (bias != nullptr) {
```

Comment:
nitpick:
1. Technically we should look at the mask to determine whether grad_bias/grad_weight is needed or not.
2. Can we put a note here that it is intentional to keep commented code in.